### PR TITLE
Add Result.isSuccess/isFailure

### PR DESCRIPTION
**Pull request by Synesso**

### CODE REVIEW

File: lib/src/main/kotlin/app/cash/quiver/extensions/Result.kt [hunk 1]
```diff
@@ -80,3 +79,15 @@ inline fun <A, B> Result<A>.tap(f: (A) -> B): Result<A> = this.map { a ->
 inline fun <A> Result<A>.flatTap(f: (A) -> Result<Any>): Result<A> = this.flatMap { a ->
   f(a).map { a }
 }
+
+/**
+ * Returns false if Success or returns the result of the given predicate to the Failure value.
+ */
+fun <T> Result<T>.isFailure(predicate: (Throwable) -> Boolean): Boolean =
+  fold(onFailure = predicate, onSuccess = { false })
+
```
Comment (cwmyers) [inline lib/src/main/kotlin/app/cash/quiver/extensions/Result.kt:88]: Can you remind me if the other types follow this pattern?

Comment (cwmyers) [reply lib/src/main/kotlin/app/cash/quiver/extensions/Result.kt:88]: I would also `inline` this method so that it is more coroutine friendly.

Comment (Synesso) [reply lib/src/main/kotlin/app/cash/quiver/extensions/Result.kt:88]: Yes, Arrow itself provides Either isLeft isRight predicate operations. This change is to support migrating from ErrorOr to Result when those operations are used.

Comment (Synesso) [reply lib/src/main/kotlin/app/cash/quiver/extensions/Result.kt]: added inline

**Review by cwmyers — APPROVED** [review]

**Pull request merged**